utils/zerovec/src/ule/plain.rs

```diff
@@ -25,6 +25,18 @@ macro_rules! impl_byte_slice_size {
             pub fn as_bytes(&self) -> &[u8] {
                 &self.0
             }
+
+            pub const fn slice_from_byte_slice(bytes: &[u8]) -> Result<&[Self], ZeroVecError> {
```

Comment:
Suggestion: We should make a name for this that makes it clear why this function exists instead of calling `parse_byte_slice`.

I was also thinking that since the main/only use case for this is to process static bytes, we could parameterize this on `N` and make it work on arrays.

Reply:
Actually, how about if we make it return a ZeroSlice and name the function something with that in the name

Reply:
We can't do that, because `ZeroSlice` is parametrized over the AsULE type

Reply:
I can't think of a name that's clearer than this; I added docs, but overall this is a weird place for the method to be and the only reason it's here is because of `const` constraints. It looks fishy because it _is_ fishy, this method does not actually belong here.

Reply:
The ZeroSlice can be parameterized on `$unaligned`.

If we don't already have a function on ZeroSlice that casts `ZeroSlice<U>` to `ZeroSlice<T>` where `U::ULE == T::ULE`, we should add one.

Reply:
I commented in chat but there's no `$unaligned` here and there's a pair of non-ULE types for every ULE type, it's somewhat of a mess.